services/server/test/helpers/ServerFixture.ts

```diff
@@ -71,8 +92,41 @@ export class ServerFixture {
         },
       });
 
-      services["initialize"]();
-      this._server = new Server(options.port);
+      await storageService.init();
+      this._sourcifyDatabase = storageService.services[
+        StorageIdentifiers.SourcifyDatabase
+      ] as SourcifyDatabaseService;
+
+      this._server = new Server(options.port!, supportedChainsMap, {
+        enabledServices: {
+          read: options.read,
+          writeOrWarn: options.writeOrWarn,
+          writeOrErr: options.writeOrErr,
+        },
+        repositoryV1ServiceOptions: {
+          ipfsApi: process.env.IPFS_API as string,
+          repositoryPath: config.get("repositoryV1.path"),
+          repositoryServerUrl: config.get("repositoryV1.serverUrl") as string,
+        },
+        repositoryV2ServiceOptions: {
+          ipfsApi: process.env.IPFS_API as string,
+          repositoryPath: config.has("repositoryV2.path")
+            ? config.get("repositoryV2.path")
+            : undefined,
+        },
+        sourcifyDatabaseServiceOptions: {
+          postgres: {
+            host: process.env.SOURCIFY_POSTGRES_HOST as string,
+            database: process.env.SOURCIFY_POSTGRES_DB as string,
+            user: process.env.SOURCIFY_POSTGRES_USER as string,
+            password: process.env.SOURCIFY_POSTGRES_PASSWORD as string,
+            port: parseInt(process.env.SOURCIFY_POSTGRES_PORT),
+          },
+        },
+      });
+
+      await this._server.services.init();
```

Comment:
Shouldn't have we removed the above `new StorageService()` part? The `new Server` part should cover initializing a new StorageService no? We're passing the exact same options to both

Reply:
I think we can remove that in a future PR, we can actually use directly a dedicated sql pool for the tests

Reply:
I'm talking about these lines specifically: 
https://github.com/ethereum/sourcify/blob/4b1874e30a819872953650f64a4fb3b7c687b38f/services/server/test/helpers/ServerFixture.ts#L69-L99

Why are they there? `storageService` is being initialized here and then from the initialized services we extract the DB to assign it to 

https://github.com/ethereum/sourcify/blob/4b1874e30a819872953650f64a4fb3b7c687b38f/services/server/test/helpers/ServerFixture.ts#L96-L99

1. This `this._sourcifyDatabase` is not used anywhere
2. It falsely gives the impression that this `_sourcifyDatabase` is the one of the `ServerFixture` but in fact it's completely separate. The `ServerFixture`'s DB is being initialized at the end:

https://github.com/ethereum/sourcify/blob/4b1874e30a819872953650f64a4fb3b7c687b38f/services/server/test/helpers/ServerFixture.ts#L128-L129